code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm

```diff
@@ -690,4 +690,23 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/airalarm, 27)
 	update_appearance()
 	update_name()
 
+//Select a random mode and switch to it
+/obj/machinery/airalarm/proc/select_random_mode()
+	var/list/possible_random_modes = list(
+		/datum/air_alarm_mode/filtering,
+		/datum/air_alarm_mode/contaminated,
+		/datum/air_alarm_mode/draught,
+		/datum/air_alarm_mode/refill,
+		/datum/air_alarm_mode/cycle,
+		/datum/air_alarm_mode/siphon,
+		/datum/air_alarm_mode/panic_siphon,
+		/datum/air_alarm_mode/off,
+		/datum/air_alarm_mode/flood,
```

Comment:
flood and vent siphon are usually locked behind emagging it fyi

Reply:
ag good catch thanks